server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -61,15 +74,65 @@ func (w *Worker) Work(ctx workflow.Context) {
 
 		msg := w.Queue.Pop()
 
-		ctx := workflow.WithValue(ctx, context.SHAKey, msg.Revision)
-		ctx = workflow.WithValue(ctx, context.DeploymentIDKey, msg.ID)
+		ctx := workflow.WithValue(ctx, internalContext.SHAKey, msg.Revision)
+		ctx = workflow.WithValue(ctx, internalContext.DeploymentIDKey, msg.ID)
+
+		// This should only happen on startup
+		// If we fail to fetch the latest deployment, log and exit the workflow
+		if w.LatestDeployment == nil {
+			w.LatestDeployment, err = w.fetchLatestDeployment(ctx, msg)
+			if err != nil {
+				logger.Error(ctx, fmt.Sprint("Unable to fetch latest deployment, worker is shutting down", err.Error()))
+				return
+			}
+		}
 
-		err = w.TerraformWorkflowRunner.Run(ctx, msg)
+		// TODO: Validate revision before executing the workflow
 
+		err = w.TerraformWorkflowRunner.Run(ctx, msg)
 		if err != nil {
 			logger.Error(ctx, "failed to deploy revision, moving to next one")
 		}
+
+		latestDeployment, err := w.persistLatestDeployment(ctx, msg)
+		if err != nil {
+			logger.Error(ctx, "failed to persist latest deploy job")
+		}
+
+		// Update the latest deployment in memory
+		w.LatestDeployment = latestDeployment
+	}
+}
+
+func (w *Worker) fetchLatestDeployment(ctx workflow.Context, deploymentInfo terraform.DeploymentInfo) (*root.DeploymentInfo, error) {
+	// Fetch latest deployment
+	var resp activities.FetchLatestDeploymentResponse
+	err := workflow.ExecuteActivity(ctx, w.DbActivities.FetchLatestDeployment, activities.FetchLatestDeploymentRequest{
+		FullRepositoryName: w.Repo.GetFullName(),
+		RootName:           deploymentInfo.Root.Name,
+	}).Get(ctx, &resp)
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching latest deployment")
+	}
+
+	return resp.DeploymentInfo, nil
+}
+
+func (w *Worker) persistLatestDeployment(ctx workflow.Context, deploymentInfo terraform.DeploymentInfo) (*root.DeploymentInfo, error) {
```

Comment:
We should be resilient to failures ideally. For example, if we can't persist, we should be able to just move onto the next deployment and persist at some point before the worker shuts down.  

Since this PR has gone through a lot of iterations already , I'd say to just add a TODO around that and we can get to it later.